packages/react-mutation-mapper/src/component/oncokb/OncoKB.tsx

```diff
@@ -86,16 +90,40 @@ export default class OncoKB extends React.Component<IOncoKbProps, {}> {
             oncoKbContent = loaderIcon('pull-left');
         } else {
             oncoKbContent = (
-                <span className={`${annotationStyles['annotation-item']}`}>
-                    <i
-                        className={annotationIconClassNames(
-                            this.props.usingPublicOncoKbInstance,
-                            this.props.indicator
+                <>
+                    {!this.props.usingPublicOncoKbInstance &&
+                        this.props.indicator && (
+                            <>
+                                {this.getAnnotationIcon(
+                                    OncoKbCardDataType.BIOLOGICAL
+                                )}
+                                {this.props.levelType?.has(
+                                    OncoKbCardDataType[OncoKbCardDataType.TXS]
+                                ) &&
+                                    this.getAnnotationIcon(
+                                        OncoKbCardDataType.TXS
+                                    )}
+                                {this.props.levelType?.has(
+                                    OncoKbCardDataType[OncoKbCardDataType.TXR]
+                                ) &&
+                                    this.getAnnotationIcon(
+                                        OncoKbCardDataType.TXR
+                                    )}
+                                {this.props.levelType?.has(
+                                    OncoKbCardDataType[OncoKbCardDataType.DX]
+                                ) &&
+                                    this.getAnnotationIcon(
+                                        OncoKbCardDataType.DX
+                                    )}
+                                {this.props.levelType?.has(
+                                    OncoKbCardDataType[OncoKbCardDataType.PX]
+                                ) &&
+                                    this.getAnnotationIcon(
+                                        OncoKbCardDataType.PX
+                                    )}
```

Comment:
Can we do the check in the this.getAnnotationIcon? It will avoid the special checks for reach leveltype.